src/main/java/wolox/training/repositories/BookRepository.java

```diff
@@ -0,0 +1,8 @@
+package wolox.training.repositories;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import wolox.training.models.Book;
+
+public interface BookRepository  extends JpaRepository<Book, Long> {
+    Book findBookByAuthor(String author, Long id);
```

Comment:
Add that the output of this repository is Optional to avoid doing null checks in the future. (`Optional<Book>`)

I leave a link to have some additional information on the subject:

- [Optionals](https://www.baeldung.com/java-optional)

Reply:
I added optional util in this method